Sources/Services/AWSPolly/PollyClient.swift

```diff
@@ -481,3 +481,25 @@ extension PollyClient: PollyClientProtocol {
     }
 
 }
+
+extension PollyClient {
+    public func presignURLForSynthesizeSpeech(input: SynthesizeSpeechInput, expiration: Foundation.TimeInterval) async throws -> Foundation.URL {
+        guard let presignedURL = try await input.presignURL(config: config, expiration: expiration) else {
```

Comment:
its better to separate this into 2 lines for debugging/readability

```
let presignedURL = try await input.presignURL(config: config, expiration: expiration)
guard let url = presignedURL else {
     // throw error
} 
// use url here
```

This is better imo for 2 reasons
- if an error occurs during the function call it won't be confused with an error being thrown violating the guard that is distinct from nullity
- it makes it more readable as we separate concerns into first we populate a variable with the return of the call the input.presignURL and then we guard on nullity

Reply:
Refer to this comment: https://github.com/awslabs/aws-sdk-swift/pull/1198#discussion_r1373860445
That's actually what I had at first, but I think the error that's thrown should have a message that allows that distinction anyways.

Reply:
Oh I mean I would defer to @jbelkins, but I would prefer to read/debug this code with it on 2 separate lines. Either way the same error is thrown. But if `try await input.presignURL()` throws an error or if the guard statement fails because that same line returns a null value (but does not throw an error) the debugger would show the same exact line as having a failure despite 2 separate reasons (failed guard vs error thrown in method call itself) which is confusing

Reply:
to explain further my reasoning:

```
let x = try await function() // I make function call
guard let _x = x else {
   throw error(something unknown happen or x is null!)
}
```

`try await function()` is marked with try because it has its own reasons for throwing an error. Separating this out to 2 lines imo makes it clear to the reader that by using a `guard` statement I am adding additional conditions that arent inherent to calling the function

Reply:
Changed as suggested.